### DISCUSSION TITLE
method of refreshing 'live' tiles, e.g. traffic layer?

### DISCUSSION THREAD

**Discussion by summitcomputing** [fleaflet/flutter_map#1206] [Q&A] [answered]
I'm using FlutterMap with Google map tiles, and have included Google's ",traffic" option in the urlTemplate to get live traffic information. However, once an area has been displayed, the traffic information never seems to change - even closing the map page & re-opening it seems to re-load the previous data.  Re-starting the app seems to be the only way! I assume this is a caching issue? Is there a way to clear the tile cache or force the tiles to re-load?

      FlutterMap(
        mapController: mapController,
        options: MapOptions(
          center: _initialLatLng,
          zoom: _mapZoom,
        ),
        layers: [
          statefulMapController.tileLayer,
          TileLayerOptions(
            overrideTilesWhenUrlChanges: true, // needed?
            urlTemplate:
                'https://{s}.google.com/vt/lyrs=m,traffic&x={x}&y={y}&z={z}',
            subdomains: ['mt0', 'mt1', 'mt2', 'mt3'],
            tileProvider: NonCachingNetworkTileProvider(), //default
          ),
        ],
      ),

**Comment by ibrierley** [comment, marked as answer]
Hmm you could try changing the URL (add some irrelevant bogus arg to the end for example) with overrideTilesWhenUrlChanges: true...

eg   urlTemplate: "https://{s}.google.com/vt/lyrs=m,traffic&x={x}&y={y}&z={z}&refresh=${count}";

then increment count whenever you want an update ?

Just thinking out loud, never tried it.

**Reply by summitcomputing** [reply]
good idea - will try & let you know. I had previously tried switching traffic on/off:
'https://{s}.google.com/vt/lyrs=m${_trafficEnabled ? ',traffic' : ''}&x={x}&y={y}&z={z}'
but that didn't work, but maybe because previously-loaded URL was already cached?

**Reply by summitcomputing** [reply]
That seems to do the trick!  Using a dummy count like that in the URL must ensure that it's re-loaded each time the count changes, rather than relying on a previously-cached version.

Thanks!

**Comment by ibrierley** [comment]
Just a note, it "may" be against Googles T&Cs to not use their own code to query their API, just make sure you cover yourself if it's part of a larger project that may get released. I may be out of date on this, just want anyone in future who reads to double check themselves.

**Reply by summitcomputing** [reply]
fair point. Thanks

**Comment by ibrierley** [comment]
You're welcome, and thanks for the feedback, may help others with a similar issue.
